### ISSUE TITLE
Enhancements for bot

### ISSUE THREAD

**Issue by Vitsowdy** [Woyken/pixelplanet.fun-bot#30]
Hello again, @Woyken 
I used your bot for 7 hour, and it's really good thing.
If you can enhance bot, make that:
1. Multi account in one window.
I need to open 4* windows of Shell, and launch copy of bots. And it's hard to find account, where i need solve captcha.
2. Speed up pixel placing.
For 5 hours 4* bots were building 70x70 picture, it buided, but too slow, and ended on 70% progress. If possible, can speed up bot?

And in theory, i can help code bot, but i middle-knowledge of JS, so idk. :/

Labels: Enhancement

**Comment by Woyken** [comment]
Yeah... There are problems with those requests. Pixelplanet is calculating the timeout per IP address. This means you can run as many bots as you want on single IP, that won't help the speed, all they'll do is fight over same pixel timeout value, and the speed you get is exactly same as if you were running just 1 instance. All I'm doing is getting current timeout ant placing all pixels I can. Waiting for some time and doing that again. There's no speeding up for this.

For best results I would have to implement some way of using proxy, but that complicates captcha solving, how to use same ip on broser AND bot at the same time and keep track which is which. Keeping in mind that pixelplanet are blocking all proxy and vpn servers they find.

From time to time I'm working on picture overlay extension for browser, I'm considering adding "auto place" feature into it (would act similar to this bot). This feature would solve most problems with current bot.

**Comment by Vitsowdy** [comment]
Okay, but what if i launched 4 browsers and have 4 different cookies?

And i launched from VPN, and I don't banned right now.

Btw, if you will use proxy servers, you can go to that [proxy list](hidemy.name).

**Comment by Woyken** [comment]
That's the point, cookies don't really matter. What matters is from which IP is the bot itself making requests. And as far as I know there's no straight forward way to make a single application to use vpn, and another use another instance of vpn.

**Comment by Vitsowdy** [comment]
Ok, tommorow i will try make some connections from proxys in JS, and will send workable code, if it will work. (Sorry for my English)

**Comment by Vitsowdy** [comment]
What about that? : https://www.npmjs.com/package/https-proxy-agent
`
var url = require('url');
var WebSocket = require('ws');
var HttpsProxyAgent = require('https-proxy-agent');

// HTTP/HTTPS proxy to connect to
var proxy = process.env.http_proxy || 'http://168.63.76.32:3128';
console.log('using proxy server %j', proxy);

// WebSocket endpoint for the proxy to connect to
var endpoint = process.argv[2] || 'ws://echo.websocket.org';
var parsed = url.parse(endpoint);
console.log('attempting to connect to WebSocket %j', endpoint);

// create an instance of the `HttpsProxyAgent` class with the proxy server information
var options = url.parse(proxy);

var agent = new HttpsProxyAgent(options);

// finally, initiate the WebSocket connection
var socket = new WebSocket(endpoint, { agent: agent });

socket.on('open', function () {
  console.log('"open" event!');
  socket.send('hello world');
});

socket.on('message', function (data, flags) {
  console.log('"message" event! %j %j', data, flags);
  socket.close();
});
`

**Comment by Woyken** [comment]
Right, this is not the problem (I already seen solution on how to use proxies for requests, and tested it locally a few months ago). 
But as I mentioned before, problem still exists.
you would need to in your browser open a tab through this proxy url, let's call it proxy1. 
Then get the fingerprint, and keep it open.
Now paste fingerprint AND proxy1 url to the bot.
Now when it requests captcha you need to open the same proxy1 website and confirm for this proxy1 using bot. This would go on for all bots. If you have 6 bots, you would need 6 browser instances and 6 consoles as well as keep track which corresponds to which.

**Comment by Vitsowdy** [comment]
But what if usw anti captcha thing, and use cookies which i saved time ago?

**Comment by Woyken** [comment]
That's not an option. The captcha is requested per IP also. Doesn't matter if you set same cookies for 2 clients, you will have to solve captcha in browser using those 2 IPs

**Comment by Vitsowdy** [comment]
If i launch bot in 2 different Shell on 2 different proxy IPs with 2 different cookies and them will have anti captcha or captha solution thing (or library), it won't work anyway?

**Comment by Woyken** [comment]
I'm not sure what are you referencing by "captcha solution" as far as I know there aren't any "auto solvers" for google captcha. And I don't have access to the related browser instance from my script. I probably could start local server, somehow invoke google captcha and call back after it's solved... But that's way more work than I'm willing to put into this project

**Comment by Vitsowdy** [comment]
That thing for example: https://github.com/tsypa/anti-captcha
But for it you need pay :/

**Comment by Vitsowdy** [comment]
Ok, i will try place that code in bot
